src/conn/conn_oligarch.c

```diff
@@ -309,19 +310,21 @@ __wt_oligarch_manager_remove_table(WT_SESSION_IMPL *session, uint32_t ingest_id)
 static int
 __oligarch_get_constituent_cursor(WT_SESSION_IMPL *session, uint32_t ingest_id, WT_CURSOR **cursorp)
 {
+    WT_CONNECTION_IMPL *conn;
     WT_CURSOR *stable_cursor;
-    WT_OLIGARCH_MANAGER *manager;
     WT_OLIGARCH_MANAGER_ENTRY *entry;
+    uint64_t global_ckpt_id;
 
     const char *cfg[] = {WT_CONFIG_BASE(session, WT_SESSION_open_cursor), "overwrite", NULL, NULL};
 
-    manager = &S2C(session)->oligarch_manager;
-    entry = manager->entries[ingest_id];
+    conn = S2C(session);
+    entry = conn->oligarch_manager.entries[ingest_id];
 
     *cursorp = NULL;
 
-    if (manager->update_dhandle) {
-        manager->update_dhandle = false;
+    WT_ACQUIRE_READ(global_ckpt_id, conn->disaggregated_storage.global_checkpoint_id);
+    if (global_ckpt_id > entry->global_checkpoint_id) {
+        entry->global_checkpoint_id = global_ckpt_id;
```

Comment:
Should we update this only after we successfully refreshed?

Reply:
This is also a shared variable?

Reply:
To your first point, that's valid - I reorganised the code a little. For Peter's benefit - we discussed this offline, and this is only used by the log replay code so it's a single-threaded context.